### PR TITLE
adding MAE (L1) loss and test

### PR DESCRIPTION
**Pull request by pjreddie**
Poked around in the loss functions for https://github.com/allenai/helios/issues/12 , would be great to chat with someone before I add L2, cross entropy, etc.

### CODE REVIEW

File: helios/train/loss.py [hunk 1]
```diff
@@ -131,6 +131,59 @@ def compute(
...
+
+        Args:
+            predictions: Model predictions.
+            targets: Ground truth targets.
+            **kwargs: Additional keyword arguments.
+
+        Returns:
+            The computed loss value.
+        """
+        # TODO: How will we deal with only training with some subset of modalities? If we use passed in modalities channels dict to define which modalities is one way but using class directly implies all used
```
Comment (gabrieltseng) [inline helios/train/loss.py]: I think the plan is for unused modalities to be reflected in the mask value (`mask_value == MASK_VALUE.MISSING`)

File: helios/train/loss.py [hunk 2]
```diff
@@ -131,6 +131,59 @@ def compute(
...
+@LOSS_REGISTRY.register("mae")
+class MAELoss(Loss):
+    """Loss function for mean average error."""
+
+    @staticmethod
+    def _flatten(x: Tensor) -> Tensor:
+        return rearrange(x, "b ... d -> b (...) d")
+
+    @staticmethod
+    def _expand_and_reciprocate(t: Tensor) -> Tensor:
```
Comment (gabrieltseng) [inline helios/train/loss.py]: Is this function necessary for MAELoss? If not I think we can remove it.

File: helios/train/loss.py [hunk 3]
```diff
@@ -131,6 +131,59 @@ def compute(
         return loss
 
 
+@LOSS_REGISTRY.register("mae")
+class MAELoss(Loss):
+    """Loss function for mean average error."""
+
+    @staticmethod
+    def _flatten(x: Tensor) -> Tensor:
```
Comment (gabrieltseng) [inline helios/train/loss.py]: nit - if we will be using this function many times (which seems likely) maybe we can move it into the Loss ABC?

File: helios/train/loss.py [hunk 4]
```diff
@@ -131,6 +131,59 @@ def compute(
         return loss
 
 
+@LOSS_REGISTRY.register("mae")
+class MAELoss(Loss):
```
Comment (gabrieltseng) [inline helios/train/loss.py]: nit - this could be confusing since we will probably want to test MAE (masked autoencoding) models. Could we call this L1 loss instead?

Comment (pjreddie) [reply helios/train/loss.py]: as follow up I'm wondering for registery name and class name something like:
l1 - L1Loss
l2 - L2Loss
OR
l1 - L1Loss
mse - MSELoss
or do we want to write out full names for these?

Comment (pjreddie) [reply helios/train/loss.py]: it's very strange to me that Pytorch uses L1Loss and MSELoss like why not L2 loss? but maybe we want to stick with their convention idk

Comment (gabrieltseng) [reply helios/train/loss.py]: I am happy with l1 and l2 loss
